### PR TITLE
[WIP] - Add queues for reservation placed

### PR DESCRIPTION
**Pull request by nuzil**
### Description (*)

https://github.com/magento/inventory/issues/3012

Added dummy processor, created queue and consumer.

### CODE REVIEW

File: InventoryIndexer/Plugin/InventorySales/ScheduleAfterPlaceReservationsForSalesEvent.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+use Magento\InventorySales\Model\PlaceReservationsForSalesEvent;
+use Magento\Framework\MessageQueue\PublisherInterface;
+
+/**
+ * Invalidate Inventory Indexer after Source was enabled or disabled.
+ */
+class ScheduleAfterPlaceReservationsForSalesEvent
+{
+
+    public const TOPIC_RESERVATION_PLACED = "inventory.reservation.place";
```
Comment (maghamed) [inline InventoryIndexer/Plugin/InventorySales/ScheduleAfterPlaceReservationsForSalesEvent.php:19]: description for all constants required

File: InventoryIndexer/Plugin/InventorySales/ScheduleAfterPlaceReservationsForSalesEvent.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+     *
+     * @param PlaceReservationsForSalesEvent $subject
+     * @param void $result
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterExecute(
+        PlaceReservationsForSalesEvent $subject,
+        $result
+    ) {
+        $this->publisher->publish(self::TOPIC_RESERVATION_PLACED, []);
```
Comment (maghamed) [inline InventoryIndexer/Plugin/InventorySales/ScheduleAfterPlaceReservationsForSalesEvent.php:46]: we should send the subject here.  Stock or SalesChannel and SKU, as the handler has to have these data to check the salability status.

From that perspective, the append reservation is more suitable. But not we are not looking for easy ways

File: InventoryIndexer/Plugin/InventorySales/ScheduleAfterPlaceReservationsForSalesEvent.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+     * @param PublisherInterface $publisher
+     */
+    public function __construct(
+        PublisherInterface $publisher
+    ) {
+        $this->publisher = $publisher;
+    }
+
+    /**
+     * Schedule stock status update after resevation placed
```
Comment (maghamed) [inline InventoryIndexer/Plugin/InventorySales/ScheduleAfterPlaceReservationsForSalesEvent.php:36]: it's not a stock status update - it's a request to reserve product

File: InventoryIndexer/etc/queue_topology.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework-message-queue:etc/topology.xsd">
+    <exchange name="magento-db" type="topic" connection="db">
+        <binding id="InventoryReservationPlaceDb" topic="inventory.reservation.place" destinationType="queue" destination="inventory.reservation.place"/>
```
Comment (maghamed) [inline InventoryIndexer/etc/queue_topology.xml:10]: InventoryReservationPlaceDb - it's not exactly what we will be doing here. We are going to update Salability status. 
Reservations are being created in the main thread

File: InventoryIndexer/Model/ReservationDummy.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ */
+class ReservationDummy
+{
+    /**
+     * @param string[] $array
+     * @return void
+     */
+    public function execute($array)
+    {
+        return true;
```
Comment (maghamed) [inline InventoryIndexer/Model/ReservationDummy.php:21]: you may accept payload here and call `IsProductSalable` service.
That's what we need to have in indexer anyways

File: InventoryIndexer/Plugin/InventorySales/ScheduleAfterPlaceReservationsForSalesEvent.php (hunk 2 above)
Comment (novikor) [reply InventoryIndexer/Plugin/InventorySales/ScheduleAfterPlaceReservationsForSalesEvent.php:46]: See https://github.com/magento/inventory/pull/3026

**Pull request closed without merging**